modules/twinklewarn.js

```diff
@@ -1081,25 +1086,31 @@ Twinkle.warn.callback.change_category = function twinklewarnCallbackChangeCatego
 		$('select[name=sub_group]').chosen('destroy');
 	}
 
-	var value = e.target.value;
+	var value = e.target.value; // Gets destroyed in the kitchensink loop
 	var sub_group = e.target.root.sub_group;
 	sub_group.main_group = value;
 	var old_subvalue = sub_group.value;
 	var old_subvalue_re;
 	if (old_subvalue) {
-		old_subvalue = old_subvalue.replace(/\d*(im)?$/, '');
-		old_subvalue_re = new RegExp(mw.util.escapeRegExp(old_subvalue) + '(\\d*(?:im)?)$');
+		if (value === 'kitchensink') { // Exact match possible in kitchensink menu
+			old_subvalue_re = new RegExp(mw.util.escapeRegExp(old_subvalue));
+		} else {
+			old_subvalue = old_subvalue.replace(/\d*(im)?$/, '');
+			old_subvalue_re = new RegExp(mw.util.escapeRegExp(old_subvalue) + '(\\d*(?:im)?)$');
+		}
 	}
 
 	while (sub_group.hasChildNodes()) {
 		sub_group.removeChild(sub_group.firstChild);
 	}
 
+	var selected = false;
 	// worker function to create the combo box entries
-	var createEntries = function(contents, container, wrapInOptgroup) {
+	var createEntries = function(contents, container, wrapInOptgroup, val) {
+		val = typeof val !== 'undefined' ? val : value; // IE doesn't support default parameters
```

Comment:
I like to use `val = val || value`. Or just pass in `value` wherever the function is being used. But I'm probably just nitpicking :)

Reply:
Likewise, although here I did this to futureproof against other falsey values, as they could theoretically be made use of; I believe default parameters (when supported) are assigned `undefined` if not provided.  I'd sooner toss in this one line then add it everywhere just because one option needs it.